src/main/java/org/vaadin/addons/tatu/TabSheet.java

```diff
@@ -239,20 +249,58 @@ public void setSelected(String tab) {
         getElement().executeJs("this.setSelectedTab($0)", tab);
     }
 
+    /**
+     * Get selected tab index.
+     *
+     * @return Index of the tab, base 0 or -1 of no selection.
+     */
+    public int getSelectedIndex() {
+        return this.getElement().getProperty("selected", -1);
+    }
     /**
      * Set selected tab using index. This will fire TabChangeEvent. Sheet
      * attached to the tab will be shown.
      * 
      * @param index
      *            Index of the tab, base 0.
      */
-    public void setSelected(int index) {
+    public void setSelectedIndex(int index) {
```

Comment:
Breaks compatibility.

Reply:
It does, but I prefer API symmetry. Requires new major version?

Reply:
Current naming is: 
- String selected
- int selectedIndex
- Component selectedComponent

This setter name change indeed breaks the compatibility with current API, but should be an obvious one. 

Once updating API, one thing that I was also wondering is that is the `String` ID meaningful? Is there a situation where it is needed in the application or could it be moved from public API? I could think index and Component get you quite far.

Reply:
I added the old method back for backward compatibility.